internal/controllers/machinedeployment/mdutil/util_test.go

```diff
@@ -76,175 +78,209 @@ func generateDeployment(image string) clusterv1.MachineDeployment {
 			Annotations: make(map[string]string),
 		},
 		Spec: clusterv1.MachineDeploymentSpec{
-			Replicas: func(i int32) *int32 { return &i }(1),
+			Replicas: pointer.Int32(3),
 			Selector: metav1.LabelSelector{MatchLabels: machineLabels},
 			Template: clusterv1.MachineTemplateSpec{
 				ObjectMeta: clusterv1.ObjectMeta{
 					Labels: machineLabels,
 				},
-				Spec: clusterv1.MachineSpec{},
+				Spec: clusterv1.MachineSpec{
+					NodeDrainTimeout: &metav1.Duration{Duration: 10 * time.Second},
+				},
 			},
 		},
 	}
 }
 
-func generateMachineTemplateSpec(annotations, labels map[string]string) clusterv1.MachineTemplateSpec {
-	return clusterv1.MachineTemplateSpec{
-		ObjectMeta: clusterv1.ObjectMeta{
-			Annotations: annotations,
-			Labels:      labels,
+func TestMachineSetsByDecreasingReplicas(t *testing.T) {
```

Comment:
Added separate cases for the test over testing a single slice of MachineSet to make all the individual cases clear and readable.